src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -22,44 +23,58 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens
     internal class SemanticTokensHelpers
     {
         /// <summary>
-        /// Returns the semantic tokens data for a given document with an optional range.
+        /// Returns the semantic tokens data for a given document with an optional ranges.
         /// </summary>
         public static async Task<int[]> ComputeSemanticTokensDataAsync(
             ClientCapabilities capabilities,
             Document document,
-            LSP.Range? range,
+            LSP.Range[]? ranges,
             ClassificationOptions options,
             CancellationToken cancellationToken)
         {
-            var tokenTypesToIndex = SemanticTokensSchema.GetSchema(capabilities.HasVisualStudioLspCapability()).TokenTypeToIndex;
+            if (ranges is not null)
+            {
+                // Expecting sorted ranges within the document
+                Array.Sort(ranges, static (r1, r2) => r1.CompareTo(r2));
+            }
```

Comment:
I don't think this is needed anymore

Reply:
Note, if this isn't needed, then I *think* you no longer need the extension methods for the lsp types